### PR TITLE
pragha: init at 1.3.4

### PR DESCRIPTION
**Pull request by mbaeten**
###### Motivation for this change
Adding new software.

###### Things done
- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 113356` at be0641e7 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li><a href="https://gist.github.com/df450c04c5f0031d26225b1f9f9f2615">pragha</a></li></li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/audio/pragha/default.nix:84:5:
```
   |
84 |     maintainers = with maintainers; [ ];
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/applications/audio/pragha/default.nix:83:5:
```
   |
83 |     license = licenses.gpl3;
   |     ^
```
</li>
    <li>warning: unused-argument

Unused argument: `gobject-introspection`.
Near pkgs/applications/audio/pragha/default.nix:6:3:
```
  |
6 |   gobject-introspection,
  |   ^
```
</li>
  </ul>
</details>

File: pkgs/applications/audio/pragha/default.nix [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  pcre,
+  pkg-config,
+  qt5,
+  sqlite,
+  stdenv,
+  taglib,
+  which,
+  xfce,
+  zlib,
+}:
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/pragha/default.nix:29]:
```suggestion
{ lib
, dbus-glib
, desktop-file-utils
, fetchurl
, glyr
, gobject-introspection
, gst_all_1
, gtk3
, hicolor-icon-theme
, intltool
, libcddb
, libcdio
, libcdio-paranoia
, libgudev
, liblastfmSF
, libnotify
, libpeas
, makeWrapper
, pcre
, pkg-config
, qtbase
, sqlite
, taglib
, which
, xfce
, zlib
}:
```

File: pkgs/applications/audio/pragha/default.nix [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+with lib;
+stdenv.mkDerivation rec {
+
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/pragha/default.nix:33]:
```suggestion
mkDerivation rec {
```

File: pkgs/applications/audio/pragha/default.nix [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  pname = "pragha";
+  version = "1.3.4";
+
+  src = fetchurl {
+    url = "https://github.com/pragha-music-player/pragha/archive/v${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/pragha/default.nix:38]: Can we easily build from source?

File: pkgs/applications/audio/pragha/default.nix [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    hicolor-icon-theme
+    libcddb
+    libcdio
+    libcdio-paranoia
+    libgudev
+    liblastfmSF
+    libnotify
+    libpeas
+    pcre
+    qt5.qtbase
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/pragha/default.nix:64]:
```suggestion
    qtbase
```

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -23727,6 +23727,8 @@ in
 
   ncmpcpp = callPackage ../applications/audio/ncmpcpp { };
 
+  pragha = callPackage ../applications/audio/pragha { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:23730]:
```suggestion
  pragha = libsForQt5.callPackage ../applications/audio/pragha { };
```

File: pkgs/applications/audio/pragha/default.nix [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    sqlite
+    taglib
+    zlib
+  ];
+
+  postInstall = ''wrapProgram $out/bin/pragha --prefix GST_PLUGIN_SYSTEM_PATH_1_0 : "$GST_PLUGIN_SYSTEM_PATH_1_0"'';
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/pragha/default.nix:70]:
```suggestion
  postInstall = ''
    wrapProgram $out/bin/pragha --prefix GST_PLUGIN_SYSTEM_PATH_1_0 : "$GST_PLUGIN_SYSTEM_PATH_1_0"
  '';
```

File: pkgs/applications/audio/pragha/default.nix [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+  # Enable glyr
+  postFixup = ''
+    wrapProgram "$out/bin/pragha" \
+      --prefix PATH ":" ${lib.makeBinPath [
+        "$out" which  glyr
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/pragha/default.nix:76]: Please do not wrap a program multiple times.

File: pkgs/applications/audio/pragha/default.nix [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+      ]}
+  '';
+
+  meta = with lib; {
+    description = "A lightweight GTK+ music manager - fork of Consonance Music Manager";
+    homepage = "https://pragha-music-player.github.io/";
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/pragha/default.nix:83]: gpl3Only or gpl3Plus?

File: pkgs/applications/audio/pragha/default.nix [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+    maintainers = with maintainers; [ ];
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/pragha/default.nix:84]: please add yourself as a maintainer.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by mbaeten** [comment]
Thanks for your help, it's very much appreciated and I fixed everything you asked for. The archive was a src build, but for clarity, I chose fetchGitHub now. Feel free to give me more hints/suggestions on how to improve the package. Optional build deps would be nice. If I find some free time, I'll probably do that.

File: maintainers/maintainer-list.nix [hunk 10]
```diff
@@ -5866,6 +5866,12 @@
     githubId = 22836301;
     name = "Mateusz Mazur";
   };
+  mbaeten = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:5869]: Please put this into a separate commit with the message ``maintainers: add mbaeten `` and all the package related changes into a commit with the message ``pragha: init at 1.3.4``.

File: pkgs/applications/audio/pragha/default.nix [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+    pcre
+    qtbase
+    sqlite
+    taglib
+    zlib
+  ];
+
+  postInstall = ''
+    wrapProgram $out/bin/pragha \
+      --prefix PATH ":" ${lib.makeBinPath [ "$out" which  glyr ]} \
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/pragha/default.nix:75]: Would it be a good idea to move the packages from $out to $out/bin which is in PATH by default?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by mbaeten** [comment]
Fixed.

File: pkgs/applications/audio/pragha/default.nix [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+, xfce, withXfce4ui ? false
+, totem-pl-parser, withTotemPlParser ? false
+# , grilo, withGrilo ? false
+# , rygel, withRygel ? true
+}:
+
+assert withGlyr -> withLastfm;
+assert withLastfm -> withCD;
+assert withCD -> libcddb != null && libcdio != null && libcdio-paranoia != null;
+assert withMtp -> libgudev != null;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/pragha/default.nix:38]:
```suggestion
```

File: pkgs/applications/audio/pragha/default.nix [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+
+with lib;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/pragha/default.nix:40]:
```suggestion
```

File: pkgs/applications/audio/pragha/default.nix [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+  ++ optional withKeybinder keybinder3
+  ++ optional withLibnotify libnotify
+  ++ optional withLastfm liblastfmSF
+  ++ optional withGlyr glyr
+  ++ optional withLibsoup libsoup
+  ++ optional withMtp libmtp
+  ++ optional withXfce4ui xfce.libxfce4ui
+  ++ optional withTotemPlParser totem-pl-parser
+  # ++ optional withGrilo grilo
+  # ++ optionals withRygel [ rygel ]
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/pragha/default.nix:87]:
```suggestion
  ++ lib.optionals withGstPlugins [ gst-plugins-good gst-plugins-bad gst-plugins-ugly ]
  ++ lib.optionals withCD [ libcddb libcdio libcdio-paranoia ]
  ++ lib.optional withGudev libgudev
  ++ lib.optional withKeybinder keybinder3
  ++ lib.optional withLibnotify libnotify
  ++ lib.optional withLastfm liblastfmSF
  ++ lib.optional withGlyr glyr
  ++ lib.optional withLibsoup libsoup
  ++ lib.optional withMtp libmtp
  ++ lib.optional withXfce4ui xfce.libxfce4ui
  ++ lib.optional withTotemPlParser totem-pl-parser
  # ++ lib.optional withGrilo grilo
  # ++ lib.optional withRygel rygel
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by mbaeten** [comment]
Thanks again for your awesome help ;)

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 113356` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>pragha</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
pragha:
<details>


A likely typo in `CPPFLAGS` argument was found, did you mean `CFLAGS` or `CXXFLAGS`?

Near pkgs/applications/audio/pragha/default.nix:90:3:

```
   |
90 |   CPPFLAGS = [ "-DHAVE_PARANOIA_NEW_INCLUDES" ];
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/attribute-typo.md
desktop-file-utils is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/applications/audio/pragha/default.nix:61:3:

```
   |
61 |   buildInputs = with gst_all_1; [
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
Unused argument: `which`.
Near pkgs/applications/audio/pragha/default.nix:14:3:

```
   |
14 | , which
   |   ^
```
</details>
</details>

File: pkgs/applications/audio/pragha/default.nix [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+  ++ lib.optional withMtp libmtp
+  ++ lib.optional withXfce4ui xfce.libxfce4ui
+  ++ lib.optional withTotemPlParser totem-pl-parser
+  # ++ lib.optional withGrilo grilo
+  # ++ lib.optional withRygel rygel
+  ;
+
+  CPPFLAGS = [ "-DHAVE_PARANOIA_NEW_INCLUDES" ];
+
+  NIX_CFLAGS_COMPILE = "-I${gst_all_1.gst-plugins-base.dev}/include/gstreamer-1.0";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/pragha/default.nix:88]:
```suggestion
  NIX_CFLAGS_COMPILE = "-I${lib.getDev gst_all_1.gst-plugins-base}/include/gstreamer-1.0";
```

File: pkgs/applications/audio/pragha/default.nix [hunk 16]
```diff
@@ -0,0 +1,105 @@
...
+
+  postInstall = ''
+    qtWrapperArgs+=(--prefix GST_PLUGIN_SYSTEM_PATH_1_0 : "$GST_PLUGIN_SYSTEM_PATH_1_0")
+
+    install -m 444 data/${pname}.desktop $out/share/applications
+    install -d $out/share/pixmaps
+    install -m 444 data/${pname}.1 $out/share/man/man1/
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/pragha/default.nix:95]: Please use installManPage.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 113356` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>pragha</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
pragha:
<details>


A likely typo in `CPPFLAGS` argument was found, did you mean `CFLAGS` or `CXXFLAGS`?

Near pkgs/applications/audio/pragha/default.nix:86:3:

```
   |
86 |   CPPFLAGS = [ "-DHAVE_PARANOIA_NEW_INCLUDES" ];
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/attribute-typo.md
desktop-file-utils is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/applications/audio/pragha/default.nix:57:3:

```
   |
57 |   buildInputs = with gst_all_1; [
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
Unused argument: `which`.
Near pkgs/applications/audio/pragha/default.nix:14:3:

```
   |
14 | , which
   |   ^
```
</details>
</details>
